test/dd/test_state_preparation.cpp

```diff
@@ -0,0 +1,41 @@
+#include "dd/Package.hpp"
+
+#include <gtest/gtest.h>
+
+TEST(DDPackageTest, GHZStateTest) {
+  auto dd = std::make_unique<dd::Package<>>(3);
+  auto ghz = dd->makeGHZState(3);
+
+  // build state vector for (|000> + |111>) * 1/sqrt(2)
+  const dd::CVec state = {dd::SQRT2_2, 0, 0, 0, 0, 0, 0, dd::SQRT2_2};
+  const auto stateDD = dd->makeStateFromVector(state);
+
+  EXPECT_EQ(ghz, stateDD);
+}
+
+TEST(DDPackage, WStateTest) {
+  auto dd = std::make_unique<dd::Package<>>(3);
+  auto wState = dd->makeWState(3);
+
+  // build state vector for (|001> + |010> + |100>) * 1/sqrt(3)
+  const dd::CVec state = {
+      0, std::sqrt(3) / 3, std::sqrt(3) / 3, 0, std::sqrt(3) / 3, 0, 0, 0};
+  const auto stateDD = dd->makeStateFromVector(state);
+
+  EXPECT_EQ(wState, stateDD);
+}
```

Comment:
Would it be possible to make these tests a little more generic?
Specifically, test a wider range of qubits. You can create parametrised tests and test fixtures using googletest pretty easily. You can have a look at https://github.com/cda-tum/mqt-core/blob/main/test/algorithms/test_wstate.cpp for example.
Would be nice to test an equal range of qubits here.
It is quite straight-forward to formulate the checks for the correct state in a generic way
- for the GHZ state the amplitudes of the 0...0 and 1...1 both have to be 1/sqrt(2)
- for the W-state the amplitudes of all states with exactly one 1 entry must have the same amplitude (should be `sqrt(n)/n` if I am not mistaken)

In addition to the above, it would be nice to extend the tests
- https://github.com/cda-tum/mqt-core/blob/main/test/algorithms/test_wstate.cpp
- https://github.com/cda-tum/mqt-core/blob/main/test/algorithms/test_entanglement.cpp

with an additional test that makes sure the DD generate by simulating the algorithm is the same as the DD generated by the newly added routines. For that, you can just 
- construct the algorithm
- construct a DD package
- construct a zero state
- call the simulate method (from https://github.com/cda-tum/mqt-core/blob/be7c5eca7d22e285388ffb62c4478b8edb97569c/include/dd/Simulation.hpp#L10-L32), yielding a DD `e`
- construct the respective state using the new routine, yielding a DD `f`
- assert that `e == f` (pointers and edge weights should match).

Reply:
I have added the requested changes, please let me know if there are any changes needed! 😃